src/ctelnet.cpp

```diff
@@ -1621,13 +1621,68 @@ void cTelnet::setGMCPVariables(const QByteArray& msg)
             return;
         }
 
-        QString version = transcodedMsg.section(QChar::LineFeed, 0);
-        // Cannot use QLatin1String(...) as that is only introduced in Qt 5.11:
-        version.remove(QStringLiteral("Client.GUI "));
-        version.replace(QChar::LineFeed, QChar::Space);
-        version = version.section(QChar::Space, 0, 0);
+        // Mudlet supports two formats for parsing data associated with
+        // Client.GUI package:
+        //
+        // JSON:       Client.GUI = {"version": "1", "url": "https://example.com"}
+        // Raw Telnet: Client.GUI <version>\n<url>
+        //
+        // If the data does not parse as JSON, we'll try Raw telnet.
+
+        QString version;
+        bool hasVersion = false;
```

Comment:
You can, I think, dispense with the `bool`-ean flags by making use of the `QString::isEmpty()` checks on the new line `1682` ...